brainiak/reprsimil/brsa.py

```diff
@@ -248,7 +251,8 @@ def fit(self, X, design, scan_onsets=None, coords=None,
         assert_all_finite(design)
         assert design.ndim == 2,\
             'The design matrix should be 2 dimension ndarray'
-        assert np.all(np.std(design, axis=0) >= 0) or self.pad_DC,\
+        assert (not np.all(np.std(design, axis=0) > 0) and self.pad_DC)\
```

Comment:
Javier would prefer to use explicit exceptions for input checking instead of asserts. We discussed this face to face. There are advantages to doing that (too long discussion for PR comments). There are also advantages to using asserts. We can leave the code as it is.